### PR TITLE
Add hotStart for P2 finite element (Issue #697)

### PR DESCRIPTION
**Pull request by wacyyang**
In this pull request, we will solve the problem of reading a P2 finite element function from h5+xmf file into Proteus.

### CODE REVIEW

**Comment by wacyyang** [comment]
To check this implementation, I am running cylinder problem 2 times. The **hotStart** is used in the 2nd run. The time interval for the first run is [0,0.2], and for the 2nd run is [0,0.1]+[0.1,0.2]. The results show a big difference in _p_ and _u_ even at _t=1.01_ after one time step. The same difference is observed for the P1P1 case also.  Maybe it is related to the _Spinup_ process.

**Comment by wacyyang** [comment]
I realized a possible reason for the failure of using HotStart in the above parallel case. It is because the data is saved using 1 processor. If I choose to save the data using 2 processors, and use 2 processors to apply HotStart, then there is no inconsistency in reading data. And it seems this piece of code works for all finite element spaces, and maybe we can move it into the base class.

File: proteus/FemTools.py [hunk 1]
```diff
@@ -5398,6 +5398,18 @@ def writeFunctionHeaderEnsight(self,u,filename,append=False,firstVariable=True,c
                                   u.name+' '+filename+u.name+'.scl****\n')
     def writeFunctionXdmf(self,ar,u,tCount=0,init=True):
         self.XdmfWriter.writeFunctionXdmf_C0P2Lagrange(ar,u,tCount=tCount,init=init)
+    def readFunctionXdmf(self,ar,u,tCount=0):
+#         import pdb
```
Comment (cekees) [inline proteus/FemTools.py]: remove these commented debugging lines

File: proteus/NumericalSolution.py [hunk 2]
```diff
@@ -1185,8 +1185,10 @@ def calculateSolution(self,runName):
             if time >= self.tnList[-1] - 1.0e-5:
                 logEvent("Modifying time interval to be tnList[-1] + tnList since tnList hasn't been modified already")
                 ndtout = len(self.tnList)
-                dtout = (self.tnList[-1] - self.tnList[0])/float(ndtout-1)
-                self.tnList = [time + i*dtout for i in range(ndtout)]
+                #dtout = (self.tnList[-1] - self.tnList[0])/float(ndtout-1)
```
Comment (cekees) [inline proteus/NumericalSolution.py]: remove comments

File: proteus/default_so.py [hunk 3]
```diff
@@ -30,3 +30,5 @@
 
 dt_system_fixed = None
 """A system-wide wide time step used by SplitOperator objects"""
+
+skipSpinupOnHotstart = False
```
Comment (cekees) [inline proteus/default_so.py]: add a doc string to this option

**Pull request merged**